lldb-eval/parser.cc

```diff
@@ -1780,7 +1816,102 @@ ExprResult Parser::ParseIntegerLiteral(clang::NumericLiteralParser& literal,
       CreateValueFromAPInt(target_, llvm::APSInt(raw_value, is_unsigned),
                            target_.GetBasicType(type));
 
-  return std::make_unique<LiteralNode>(std::move(value));
+  return std::make_unique<LiteralNode>(token.getLocation(), std::move(value));
+}
+
+// Parse a builtin_func.
+//
+//  builtin_func:
+//    builtin_func_name "(" {builtin_func_argument_list} ")"
+//
+//  builtin_func_name:
+//    "__log2"
+//
+//  builtin_func_argument_list:
+//    builtin_func_argument
+//    builtin_func_argument_list "," builtin_func_argument_list
+//
+//  builtin_func_argument:
+//    expression
+//
+ExprResult Parser::ParseBuiltinFunction(
+    std::unique_ptr<BuiltinFunctionDef> func_def) {
+  // Check for builtin debugger intrinsic function.
+  clang::SourceLocation loc = token_.getLocation();
+  ConsumeToken();
+
+  Expect(clang::tok::l_paren);
+  ConsumeToken();
+
+  std::vector<ExprResult> arguments;
+
+  do {
+    // Eat the comma if this is not the first iteration.
+    if (arguments.size() > 0) {
```

Comment:
Nit: I guess you could consume the token unconditionally and remove the ConsumeToken for lparen.

Reply:
This is no longer the case when handling empty argument list